cogs/Voting.py

```diff
@@ -0,0 +1,644 @@
+    async def cancelVote(self, ctx):
+        """
+        Cancels a vote you are setting up
+        """
+        self.vote_manager.cancel_vote(ctx.author.id)
+        await ctx.send("Your active vote has been cancelled")
+
+    @is_vote_caller()
+    @vote.command(name="send")
+    async def sendVote(self, ctx):
+        """
+        Sends a vote to all users within the restrictions set with the current options added
+        """
+        vote = self.vote_manager.get_vote(ctx)
+        if not vote.is_ready():
+            await ctx.send("Please add more than 1 option to vote for")
+            return
+
+        users = [x for x in ctx.guild.members if not x.bot]
+        if vote.target_voice_channel:
+            print(vote.target_voice_channel)
+            vc_users = discord.utils.get(ctx.guild.voice_channels, id=vote.target_voice_channel).members
+            print(vc_users)
+            users = list(set(vc_users) & set(users))
+        if vote.target_roles:
+            role_users = []
+            for role in vote.target_roles:
+                role_users += role.members
+            role_users = list(dict.fromkeys(role_users))
+            users = list(set(role_users) & set(users))
+        for user in users:
+            msg = await user.send(f"You have been asked to participate in this vote from {ctx.guild.name}.\nPlease react to make your choice (You can change your mind until the vote is closed)", embed=create_embed(vote))
+            vote.register_sent(user.id, msg.id)
+            await add_reactions(vote, msg)
+        await ctx.send(f"Sent vote to {len(users)} users")
+
+    @is_vote_caller()
+    @vote.command(name="close")
+    async def close(self, ctx):
+        """
+        Ends a vote, and collects the results
+        """
+        vote = self.vote_manager.get_vote(ctx)
+        results = await get_results(self.bot, vote)
+        self.vote_manager.cancel_vote(ctx.author.id)
+        embed = await make_result_embed(vote, results)
+        if vote.chair:
+            chair = await self.bot.fetch_user(vote.chair)
+            await chair.send(embed=embed)
+            await ctx.send(f"Sent results to {chair}")
+        else:
+            await ctx.send(embed=embed)
+
+    @is_vote_caller()
+    @vote.command(name="checkResults")
+    async def check(self, ctx):
+        """
+        Checks the results of a vote without closing it
+        """
+        vote = self.vote_manager.get_vote(ctx)
+        results = await get_results(self.bot, vote)
+        embed = await make_result_embed(vote, results)
+        await ctx.send(embed=embed)
+
+    @vote.command(name="testVote")
+    async def testVote(self, ctx):
+        # vote setup for ease of testing
+        vote = self.vote_manager.create_vote(ctx, "Test")
+        vote.add_option(Option("test1", "test1"))
+        vote.add_option(Option("test2", "test2"))
+        vote.set_vc(788169900710821891)
+
+    async def update_vote_message(self, payload):
```

Comment:
Everything in payload should be split in the method using this function. This should instead have params message_id, user_id